packages/app-page-builder/src/editor/plugins/elements/block/Block.tsx

```diff
@@ -16,22 +18,40 @@ interface BlockType {
 }
 const Block: React.FC<BlockType> = ({ element }) => {
     const { id } = element;
+    const { getElementTree } = useEventActionHandler();
+    const [elementTree, setElementTree] = useState<PbElement | null>(null);
+
+    useEffect(() => {
+        async function getBlockElementTree() {
+            if (!element) {
+                return;
+            }
+            const tree = (await getElementTree(element)) as PbElement;
+            setElementTree(tree);
+        }
+
+        getBlockElementTree();
+    }, [element, setElementTree]);
 
     return (
         <BlockStyle id={id} style={{ position: "relative" }}>
             <ElementAnimation>
                 <ElementRoot element={element}>
-                    {({ elementStyle, elementAttributes, customClasses, combineClassNames }) => (
-                        <BlockContainer
-                            elementId={id}
-                            elementStyle={elementStyle}
-                            elementAttributes={elementAttributes}
-                            customClasses={[
-                                "webiny-pb-layout-block webiny-pb-base-page-element-style",
-                                ...customClasses
-                            ]}
-                            combineClassNames={combineClassNames}
-                        />
+                    {element.data?.blockId ? (
+                        <RenderElement key={element.id} element={elementTree} />
+                    ) : (
+                        ({ elementStyle, elementAttributes, customClasses, combineClassNames }) => (
+                            <BlockContainer
+                                elementId={id}
+                                elementStyle={elementStyle}
+                                elementAttributes={elementAttributes}
+                                customClasses={[
+                                    "webiny-pb-layout-block webiny-pb-base-page-element-style",
+                                    ...customClasses
+                                ]}
+                                combineClassNames={combineClassNames}
+                            />
+                        )
```

Comment:
In [this branch](https://github.com/webiny/webiny-js/blob/page-builder-blocks-v2-disable-block-element/packages/app-page-builder/src/render/components/ElementRoot.tsx#L115), I made `ElementRoot` composable, and using that you can nicely solve this via a plugin specific to the `pageEditor`.

See an example here: https://github.com/webiny/webiny-js/blob/page-builder-blocks-v2-disable-block-element/packages/app-page-builder/src/render/components/ElementRoot.tsx#L115

It was related to `Disable editing of blocks inside Page Builder` task, and I wrote a full description with links in Notion, but now I'm not sure you have access to that 🤔 In any case, please merge that branch into your working branch and you'll get these mechanisms of composing the `ElementRoot`.